chore: fix prettier, move pages, add husky

isabellabrookes opened the pull request:
- Added husky for pre-commit hooks
- Updated documentation
- Moved pages to their own subdir `_pages/` to clean up root
- Ran prettier on all files
- Added GHA for formatting via prettier on pull-request

thisislawatts commented:
Praise: Thanks for introducing Prettier @isabellabrookes that is a really positive change ✨ 

Suggestion (non-blocking): Perhaps it would be worth adopting a GitHub action to lint the changes to avoid regressions being introduced by folks who opt of git hooks

b-n commented:
@isabellabrookes do you feel strongly about precommit hooks? I have a preference for merge checks vs. precommit.

Happy to describe the reasons why, but shortcutting the conversation by asking the question first.

I'm happy to make/push a Github Action for prettier as a pre merge check since I'm asking for the alternative.

b-n commented:
@isabellabrookes Opt 3 has my preference. LMK if I can help with anything, and much appreciation for this work!

b-n commented on _includes/events.html:
```diff
@@ -1,21 +1,15 @@
-      {% include event-card.html %}
-    {% endfor %}
-  {% endif %}  
+  {% assign events = site.events | filter_tags: include.tags | reverse %} {%
+  assign future_events = events | where_exp: "event", "event.date >= site.time"
+  %} {% if include.limit and future_events.size > 0 %} {% for event in
+  future_events reversed limit: include.limit %} {% include event-card.html %}
+  {% endfor %} {% else %} {% for event in events limit: include.limit %} {%
+  include event-card.html %} {% endfor %} {% endif %}
```
This looks a little weird now. I'm wondering whether prettier can handle these erb templates, or whether we can refactor this. I think we still merge for now and can come back to it later.

isabellabrookes replied:
I needed to add an override so the prettier liquid plugin could apply to .html extensions

b-n commented on _includes/head.html:
```diff
@@ -1,18 +1,29 @@
-  <link rel="alternate" type="application/rss+xml" title="{{ site.title | escape }}" href="{{ "/feed.xml" | relative_url }}">
-  <link rel="shortcut icon" href="/assets/favicon.png" type="image/x-icon">
-  <link rel="icon" href="/assets/favicon.png" type="image/x-icon">
+  <link
+    rel="alternate"
+    type="application/rss+xml"
+    title="{{ site.title | escape }}"
+    href="{{ 'feed.xml' | relative_url }}"
+  />
```
This is why prettier is so nice <3

b-n commented on _includes/main.js:
```diff
@@ -1,217 +1,223 @@
 let exampleEmployeeCount = Math.floor(Math.random() * (1000 - 25 + 1)) + 25;
 
-function onLoad(){
+function onLoad() {
```
Not for now, but I think we can :x: this function in a later PR (along with other functions here).

isabellabrookes replied:
Pretty much this entire file is unused. I was getting warnings from my IDE

b-n commented on _layouts/translated.html:
```diff
@@ -1,17 +1,16 @@
-{% endif %}
-{{ content }}
+<div align="right">
+  {% assign languages = page.languages | default: site.languages %} {% for lang
+  in languages %} {% if languages.size == 1 %} {% elsif site.lang == lang %}
```
This one we can revisit in another PR too - there has to be another way for ERB to make it easier to maintain etc.

isabellabrookes replied:
I needed to add an override so the prettier liquid plugin could apply to .html extensions

b-n commented on docker-compose.yml:
```diff
@@ -1,7 +1,7 @@
   jekyll:
     image: jekyll/jekyll:4 # Same version as in the Gemfile
-    command: jekyll serve --watch --incremental
+    command: jekyll serve --watch --incremental --livereload
```
<3

b-n commented on docs/contributing.md:
```diff
@@ -0,0 +1,49 @@
+# Contributing
```
:chart_with_upwards_trend: :brain:

b-n commented on events.json:
```diff
@@ -1,12 +0,0 @@
----
```
Hrmmmm. Site still works without this?

There is an admin page here: https://techwerkers.nl/admin/ and I'm wondering if this file is needed there,

isabellabrookes replied:
Honestly, I yolo-ed. It didn't break my the docker thingy. I don't have access to netlify, so I'm not sure it's something I can check. Can we use the preview: https://deploy-preview-14--twc-site-nl.netlify.app/admin ?

b-n commented on events.yml:
```diff
@@ -1,12 +0,0 @@
----
```
As previous comment too

b-n reviewed (changes requested):
Whilst here I thought I'd check file by file (otherwise I'm a hypocrit at work :sweat_smile:).

"Request changes" per the GHA discussion.

There is a question around events.[yaml|json] I have :thinking: 

The other comments are mostly just so I remember to make issues. Found this along the way: https://github.com/adamzapasnik/prettier-plugin-erb (for a new PR though)

isabellabrookes commented:
@b-n & @thisislawatts
1. fixed the liquid/erb issue
2. updated the docs with _all_ the options for pre-commit hooks/prettier
3. added a GHA that runs prettier, and commits any changes for you. Tested successfully here - ([2261c9a](https://github.com/techworkersco/twc-site-nl/pull/14/commits/2261c9ad8e99db57cec60a49ad51afeb7d0ceb60))

As for the events.json/yml - I have no clue as I can't access netlify. I can put them back easily enough

b-n commented on .github/workflows/pull-request.yaml:
```diff
@@ -0,0 +1,38 @@
+name: pull-request
```
```suggestion
name: lint
```

b-n commented on .github/workflows/pull-request.yaml:
```diff
@@ -0,0 +1,38 @@
+      - name: Install dependencies
+        run: npm install
+
+      - name: Prettify code
+        run: npx prettier . --write
+
+      - name: Commit changes
+        uses: EndBug/add-and-commit@v9
+        with:
+          message: 'gha(prettier): format files'
```
```suggestion
      - name: Prettify code
        run: npx prettier . --check
```

b-n replied:
Per the name change suggestion above. Happy to debate this too though.

This comes from my general usage of github actions in that the usage i've seen is either static analysis (linting/type checking) or it's used for more broader CI/CD things (causing net effects on other repos etc). For me, it feels like commiting on top of peoples work can feel a little black magic-y  (for people that are new to the repo), and it causes some more work for the developer if they're wanting to keep their commits clean.

Thoughts?

b-n replied:
@thisislawatts Maybe you have opinions here too?

isabellabrookes replied:
With branch protection, I'm happy to remove the git commit/push

b-n reviewed (changes requested):
One last little bit to discuss with the action. The rest of the changes are :white_check_mark:

b-n commented:
:shipit:
